xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-webjar/src/main/webjar/lightbox.js

```diff
@@ -0,0 +1,293 @@
+    var url = new URL(imageSrc);
+    var searchParams = new URLSearchParams(url.search);
+    searchParams.delete('width');
+    searchParams.delete('height');
+    url.search = searchParams.toString();
+    return url.toString();
+  };
+
+  /**
+   * Scale image to thumbnail. This may not be applied for external urls.
+   */
+  var createThumbnailURL = function(imageSrc) {
+    var url = new URL(imageSrc);
+    var searchParams = new URLSearchParams(url.search);
+    searchParams.append('width', '150');
+    searchParams.append('height', '150');
+    url.search = searchParams.toString();
+    return url.toString();
+  };
+
+  /**
+   * Hide or display the lightbox description, considering the caption.
+   */
+  var toggleDescription = function(display, withCaption) {
+    if (display) {
+      $('.lightboxDescription').css('display', 'block');
+      if (withCaption) {
+        $('#blueimp-gallery >div.slides').css('bottom', '10%');
+        $('#blueimp-gallery .indicator').css('bottom', '12%');
+      } else {
+        $('#blueimp-gallery >div.slides').css('bottom', '5%');
+        $('#blueimp-gallery .indicator').css('bottom', '7%');
+      }
+    } else {
+      $('.lightboxDescription').css('display', 'none');
+      // Center the image.
+      $('#blueimp-gallery >div.slides').css('bottom', 0);
+      $('#blueimp-gallery .indicator').css('bottom', '2%');
+    }
+  };
+
+  /**
+   * Update lightbox description using given information.
+   */
+  var updateLightboxDescription = function(metadata, caption) {
+    $('.lightboxDescription .caption').empty();
+    $('.lightboxDescription .title').empty();
+    $('.lightboxDescription .publisher').empty();
+    $('.lightboxDescription .date').empty();
+
+    if (caption != undefined) {
+      $('.lightboxDescription .caption').html(caption);
+    }
+
+    if (metadata != undefined) {
+      if (metadata.name != undefined) {
+        $('.lightboxDescription .title').text(metadata.name);
+      }
+
+      if (metadata.author != undefined) {
+        $('.lightboxDescription .publisher')
+          .text(l10n.get('author', XWiki.Model.resolve(metadata.author, XWiki.EntityType.DOCUMENT).name));
+      }
+
+      if (metadata.date != undefined) {
+        $('.lightboxDescription .date').text(l10n.get('date', new Date(metadata.date).toLocaleDateString()));
+      }
+    }
+
+    var displayDescription = metadata != undefined && $('.blueimp-gallery-controls').length > 0;
+    toggleDescription(displayDescription, caption != undefined);
+  };
+
+  /**
+   * Get information for attachment, when this can be found inside the current page.
+   */
+  var getAttachmentInfo = function(imageURL) {
+    var deferred = $.Deferred();
+    if (cachedAttachments[imageURL] != undefined) {
+      deferred.resolve(cachedAttachments[imageURL]);
+    } else {
+      var serviceDoctRef = XWiki.Model.resolve('XWiki.Lightbox.GetPageAttachmentsService', XWiki.EntityType.DOCUMENT);
```

Comment:
Why ``Doct`` and not ``Doc``? Also, the service name can be better. Maybe ``XWiki.Lightbox.AttachmentMetaDataService`` or simply ``XWiki.Lightbox.AttachmentService``.